extension/encoding/awslogsencodingextension/internal/unmarshaler/vpc-flow-log/vpc_flow_log_unmarshaler.go

```diff
@@ -0,0 +1,332 @@
+			)
+		}
+	}
+
+	scopeLogs := v.getScopeLogs(*key, scopeLogsByResource)
+	rScope := scopeLogs.LogRecords().AppendEmpty()
+	record.MoveTo(rScope)
+
+	return nil
+}
+
+// getScopeLogs for the given key. If it does not exist yet,
+// create new scope logs, and add the key to the logs map.
+func (v *vpcFlowLogUnmarshaler) getScopeLogs(key resourceKey, logs map[resourceKey]plog.ScopeLogs) plog.ScopeLogs {
+	scopeLogs, ok := logs[key]
+	if !ok {
+		scopeLogs = plog.NewScopeLogs()
+		scopeLogs.Scope().SetName(metadata.ScopeName)
+		scopeLogs.Scope().SetVersion(v.buildInfo.Version)
+		logs[key] = scopeLogs
+	}
+	return scopeLogs
+}
+
+// handleField analyzes the given field and it either
+// adds its value to the resourceKey or puts the
+// field and its value in the attributes map. If the
+// field is not recognized, it returns false.
+func handleField(field string, value string, record plog.LogRecord, key *resourceKey) (bool, error) {
+	// convert string to number
+	getNumber := func(value string) (int64, error) {
+		n, err := strconv.ParseInt(value, 10, 64)
+		if err != nil {
+			return -1, fmt.Errorf("%q field in log file is not a number", field)
+		}
+		return n, nil
+	}
+
+	// convert string to number and add the
+	// value to an attribute
+	addNumber := func(field, value, attrName string) error {
+		n, err := getNumber(value)
+		if err != nil {
+			return fmt.Errorf("%q field in log file is not a number", field)
+		}
+		record.Attributes().PutInt(attrName, n)
+		return nil
+	}
+
+	switch field {
+	// TODO Add support for ECS fields
+	case "account-id":
+		key.accountID = value
+	case "vpc-id":
+		record.Attributes().PutStr("aws.vpc.id", value)
+	case "subnet-id":
+		record.Attributes().PutStr("aws.vpc.subnet.id", value)
+	case "instance-id":
+		record.Attributes().PutStr(conventions.AttributeHostID, value)
+	case "az-id":
+		record.Attributes().PutStr("aws.az.id", value)
+	case "interface-id":
+		record.Attributes().PutStr("aws.eni.id", value)
+	case "pkt-srcaddr":
+		record.Attributes().PutStr(conventions.AttributeSourceAddress, value)
+	case "pkt-dstaddr":
+		record.Attributes().PutStr(conventions.AttributeDestinationAddress, value)
+	case "srcport":
+		if err := addNumber(field, value, conventions.AttributeSourcePort); err != nil {
+			return false, err
+		}
+	case "dstport":
+		if err := addNumber(field, value, conventions.AttributeDestinationPort); err != nil {
+			return false, err
+		}
+	case "protocol":
+		n, err := getNumber(value)
+		if err != nil {
+			return false, err
+		}
+		name, ok := protocolNames[int(n)]
+		if !ok {
+			return true, fmt.Errorf("number %d does not correspond to a IANA protocol number", n)
+		}
+		record.Attributes().PutStr(conventions.AttributeNetworkProtocolName, name)
+	case "type":
+		record.Attributes().PutStr(conventions.AttributeNetworkType, strings.ToLower(value))
+	case "region":
+		key.region = value
+	case "flow-direction":
+		record.Attributes().PutStr(conventions.AttributeNetworkIoDirection, value)
```

Comment:
We should translate the value to the SemConv expected value here: egress -> transmit, ingress -> receive.

Reply:
Done.